stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api2/MoshiChatApi.kt

```diff
@@ -3,19 +3,41 @@ package io.getstream.chat.android.client.api2
 import io.getstream.chat.android.client.api.ChatApi
 import io.getstream.chat.android.client.api2.mapping.toDomain
 import io.getstream.chat.android.client.api2.mapping.toDto
+import io.getstream.chat.android.client.api2.model.dto.DeviceDto
+import io.getstream.chat.android.client.api2.model.dto.ReactionDto
+import io.getstream.chat.android.client.api2.model.requests.AddDeviceRequest
 import io.getstream.chat.android.client.api2.model.requests.MessageRequest
+import io.getstream.chat.android.client.api2.model.requests.ReactionRequest
 import io.getstream.chat.android.client.call.Call
 import io.getstream.chat.android.client.call.map
+import io.getstream.chat.android.client.logger.ChatLogger
+import io.getstream.chat.android.client.models.Device
 import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.Reaction
 
 internal class MoshiChatApi(
     private val apiKey: String,
     private val legacyApiDelegate: ChatApi,
     private val messageApi: MessageApi,
+    private val deviceApi: DeviceApi,
 ) : ChatApi by legacyApiDelegate {
 
-    private lateinit var userId: String
-    private lateinit var connectionId: String
+    val logger = ChatLogger.get("MoshiChatApi")
+
+    private var userId: String = ""
+        get() {
+            if (field == "") {
+                logger.logE("userId accessed before being set")
+            }
+            return field
+        }
+    private var connectionId: String = ""
+        get() {
+            if (field == "") {
+                logger.logE("connectionId accessed before being set")
+            }
+            return field
+        }
```

Comment:
Using `lateinit` caused a crash here due to a bug in our client initialization code. This is a temporary workaround to make the Moshi implementation testable.